cdn/signedcookies/sign_cookie_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestReadKeyFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "cdnkey")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	f.Close()
+
+	key := `nZtRohdNF9m3cKM24IcK4w==`
+	expected := []byte{0x9d, 0x9b, 0x51, 0xa2, 0x17, 0x4d, 0x17, 0xd9,
+		0xb7, 0x70, 0xa3, 0x36, 0xe0, 0x87, 0x0a, 0xe3}
+	if err := ioutil.WriteFile(f.Name(), []byte(key), 0600); err != nil {
+		t.Fatal(err)
+	}
+	b, err := readKeyFile(f.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(b, expected) {
+		t.Fatalf("got=%v; expected=%v", b, expected)
+	}
+}
+
+func TestSignCookie(t *testing.T) {
+	testKey := []byte{0x9d, 0x9b, 0x51, 0xa2, 0x17, 0x4d, 0x17, 0xd9,
+		0xb7, 0x70, 0xa3, 0x36, 0xe0, 0x87, 0x0a, 0xe3} // base64url: nZtRohdNF9m3cKM24IcK4w==
+
+	cases := []struct {
+		testName   string
+		urlPrefix  string
+		keyName    string
+		key        []byte
+		expiration time.Time
+		out        string
+		shouldPass bool
```

Comment:
Maybe `shouldMatch`? Wasn't clear if this was for errors or something else at first.